show_client/mac_cli.go

```diff
@@ -46,3 +48,113 @@ func getMacAgingTime(options sdc.OptionMap) ([]byte, error) {
 	}
 	return json.Marshal(result)
 }
+
+// MacEntry represents a single FDB entry
+type MacEntry struct {
+	Vlan       string `json:"vlan"`
+	MacAddress string `json:"macAddress"`
+	Port       string `json:"port"`
+	Type       string `json:"type"`
+}
+
+const (
+	ApplDb   = "APPL_DB"
+	StateDb  = "STATE_DB"
+	FDBTable = "FDB_TABLE"
+)
+
+// getMacTable queries APPL_DB and STATE_DB FDB_TABLE entries and returns either the list or count per options
+func getMacTable(options sdc.OptionMap) ([]byte, error) {
+	// Parse filters
+	vlanFilter := -1
+	if v, ok := options["vlan"].Int(); ok {
+		vlanFilter = v
+	}
+	portFilter, _ := options["port"].String()
+	addrFilter, _ := options["address"].String()
+	typeFilter, _ := options["type"].String()
+	wantCount, _ := options["count"].Bool()
+
+	// Fetch APPL_DB and STATE_DB separately to preserve source and avoid key collisions
+	applData, err := GetMapFromQueries([][]string{{ApplDb, FDBTable}})
```

Comment:
I talked to @prsunny, we should not pull FDB_TABLE from APPL_DB as they may not be the correct changes as they are only populated in warmboot cases. Please only use STATE_DB for pulling FDB entries.